lib/i2cscan/i2cscan.cpp

```diff
@@ -43,12 +50,45 @@ namespace I2CSCAN
         if(!found) {
             Serial.println("[ERR] I2C: No I2C devices found");
         }
+    // begin Reset the Wire interface back to the original Values
+    #if ESP32
+        Wire.end();
+    #endif
+        Wire.begin(static_cast<int>(PIN_IMU_SDA), static_cast<int>(PIN_IMU_SCL));
+    // end Reset the Wire interface back to the original Values
     }
 
+    boolean  inArray(uint8_t search, uint8_t search2, uint8_t arraysearch[], uint8_t size )
+    {
+        boolean ret = false;
+        
+        if (size>0) 
+        {
+            for (uint8_t k = 0; k < size; k++)
+            {
+                if (search == arraysearch[k]) 
+                {
+                    ret = true;
+                    k = size;
+                }
+                if (search2 == arraysearch[k]) 
+                {
+                    ret = true;
+                    k = size;
+                }
+            }
+        }
+        //Serial.printf("InArray: %d, %d, %d, result: %x\n", search, search2, size, ret);
+        return ret;
+    }
+    
     bool checkI2C(uint8_t i, uint8_t j)
     {
         bool found = false;
-        Wire.begin(portArray[i], portArray[j]);
+#if ESP32
```

Comment:
Shouldn't this be for the ESP32C3?

Reply:
No, it is for ESP32 (espressif32). They did add a check on the new versions (i think from 5.1 on)